src/persist/src/operators/source.rs

```diff
@@ -147,8 +157,18 @@ where
                             activator.activate();
                         }
                         ListenEvent::Sealed(ts) => {
-                            cap.downgrade(&ts);
-                            activator.activate();
+                            // We only need to check that the incoming notifications
+                            // are in advance of the snapshot's lower bound because the
+                            // snapshot might have some overlap with the notifications.
+                            // The seals are required to be in order and so we don't
+                            // need to check that each seal is in advance of its
+                            // predecessor.
+                            if lower_filter.less_equal(&ts) {
+                                cap.downgrade(&ts);
+                                activator.activate();
+                            } else {
+                                debug!("Got a seal notification that was not beyond the lower snapshot filter. lower_filter: {:?}, seal ts: {:?}", lower_filter, ts);
```

Comment:
now that we've identified the bug, IMO this message isn't particularly helpful and might even be distracting. but it also probably doesn't hurt, so leave it in if you feel strongly